### ISSUE TITLE
promtool test proposal to add less strict logic for test definition

### ISSUE THREAD

**Issue by Slach** [prometheus/prometheus#13244]
### Proposal

The current approach for tests requires adding a full description and full label set from alert in rules file to alert test file, it requires copy+paste description from rules file to test file with substitution of template variables

What if I only need to just test selected alerts happens and selected labels present in alert label set?

I propose add option `--ignore-unknown-labels` and `--ignore-unknown-annotations` to `test rule` command

**Comment by alexgreenbank** [comment]
Hello from the bug scrub.

Ping @dgl do you have an opinion on this issue?

**Comment by dgl** [comment]
I don't think this request as is makes sense. Having test execution depend on command line flags isn't really desirable. It is possible to achieve something like this currently in a test by querying the `ALERTS` timeseries, rather than using alert tests. We also already have an issue for a way to avoid matching on everything in alerts, although the exact implementation needs more design thought.
